src/fruity/fruity-host-session.vala

```diff
@@ -45,6 +45,51 @@ namespace Frida {
 		}
 
 		private async void do_start () {
+			bool success = yield try_start_control_connection ();
+
+			if (success) {
+				/* perform a dummy-request to flush out any pending device attach notifications */
+				try {
+					yield control_client.connect_to_port (Fruity.DeviceId (uint.MAX), 0, start_cancellable);
+					assert_not_reached ();
+				} catch (GLib.Error expected_error) {
+					if (expected_error.code == IOError.CONNECTION_CLOSED) {
+						/* usbmuxd closes the connection when receiving commands in the wrong state */
+						control_client.close.begin (null);
+
+						success = yield try_start_control_connection ();
+						if (success) {
+							Fruity.UsbmuxClient flush_client = null;
+							try {
+								flush_client = yield Fruity.UsbmuxClient.open (start_cancellable);
+								try {
+									yield flush_client.connect_to_port (Fruity.DeviceId (uint.MAX), 0, start_cancellable);

```

Comment:
We should try to keep lines <= 140 cols. (Slowly moving frida-core towards this.)